hsds/util/chunkUtil.py

```diff
@@ -1148,41 +1273,95 @@ def chunkQuery(
+    if eval_str:
+        log.debug(f"eval_str: {eval_str}")
+    else:
+        log.debug("no eval_str")
+
+    # check for a where in statement
+    where_field = _getWhereFieldName(query)
+    if where_field:
+        log.debug(f"where_field: {where_field}")
+        if where_field not in field_names:
+            msg = f"where field {where_field} is not a member of dataset type"
+            raise ValueError(msg)
+        where_elements = _getWhereElements(query)
+        if not where_elements:
+            msg = "query: where key word with no elements"
+            raise ValueError(msg)
+        # convert to ndarray, checking than we can convert to our dtype along the way
+        try:
+            where_elements_arr = np.array(where_elements, dtype=dset_dt[where_field])
+        except ValueError:
+            msg = "where elements are not compatible with field datatype"
+            raise ValueError(msg)
+        log.debug(f"tbd: where_elements_arr; {where_elements_arr}")
+        log.debug(f"tbd: chunk_sel[{where_field}]: {chunk_sel[where_field]}")
+        isin_arr = np.isin(chunk_sel[where_field], where_elements_arr)
+        log.debug(f"tbd: isin_arr: {isin_arr}")
+
+        if not np.any(isin_arr):
+            # all false
+            log.debug("query - no rows found for where elements")
+            return None
+
+        chunk_sel = chunk_sel[isin_arr]
+        log.debug(f"tbd - chunk_sel after boolean selection: {chunk_sel}")
+
+        if len(chunk_sel) == 0:
+            log.debug("query - no elements matched where list")
+            return None
+        if not eval_str:
+            # can just return the array
+            return chunk_sel
+    elif eval_str:
+        log.debug("no where keyword")
+    else:
+        log.warn("query  - no eval and no where in, returning None")
+        return None
+
     if query_update:
-        replace_mask = [
-            None,
-        ] * len(field_names)
+        if where_field:
+            msg = "query update is not supported with where in"
+            raise ValueError(msg)
+        replace_mask = [None,] * len(field_names)
         for i in range(len(field_names)):
             field_name = field_names[i]
             if field_name in query_update:
                 replace_mask[i] = query_update[field_name]
         log.debug(f"chunkQuery - replace_mask: {replace_mask}")
-        replace_fields = [
-            None,
-        ] * len(field_names)
+        replace_fields = [None, ] * len(field_names)
         if replace_mask == replace_fields:
             msg = "chunkQuery - no fields found in query_update"
             raise ValueError(msg)
     else:
         replace_mask = None
 
-    x = chunk_arr[slices]
-    eval_str = _getEvalStr(query, "x", field_names)
+    if not eval_str:
+        if where_field:
+            # return array based on where in elements
+            log.debug("query - no eval_str, returning where in array")
+            return chunk_arr
```

Comment:
`chunk_sel` seems to be the result of the selection from a `where in`  query - should that be returned here instead?